src/sentry/sentry_metrics/indexer/base.py

```diff
@@ -428,36 +428,36 @@ def bulk_record(
             3. ids that have been rate limited (postgres/spanner)
             4. ids from newly created db records (postgres/spanner)
 
-        Each step will start off with a KeyCollection and KeyResults:
-            keys = KeyCollection(mapping)
-            key_results = KeyResults()
+        Each step will start off with a UseCaseKeyCollection and UseCaseKeyResults:
+            keys = UseCaseKeyCollection(mapping)
+            results = UseCaseKeyResults()
 
         Then the work to get the ids (either from cache, db, etc)
-            .... # work to add results to KeyResults()
+            .... # work to add results to UseCaseKeyResults()
 
         Those results will be added to `mapped_results` which can
         be retrieved
-            key_results.get_mapped_results()
+            results.get_mapped_results()
 
         Remaining unmapped keys get turned into a new
-        KeyCollection for the next step:
-            new_keys = key_results.get_unmapped_keys(mapping)
+        UseCaseKeyCollection for the next step:
+            new_keys = results.get_unmapped_keys(mapping)
 
         When the last step is reached or a step resolves all the remaining
         unmapped keys the key_results objects are merged and returned:
             e.g. return cache_key_results.merge(db_read_key_results)
         """
         raise NotImplementedError()
 
-    def record(self, use_case_id: UseCaseKey, org_id: int, string: str) -> Optional[int]:
+    def record(self, use_case_id: UseCaseId, org_id: int, string: str) -> Optional[int]:
         """Store a string and return the integer ID generated for it
 
         With every call to this method, the lifetime of the entry will be
         prolonged.
         """
         raise NotImplementedError()
 
-    def resolve(self, use_case_id: UseCaseKey, org_id: int, string: str) -> Optional[int]:
+    def resolve(self, use_case_key: UseCaseKey, org_id: int, string: str) -> Optional[int]:
```

Comment:
Maybe let's not rename `use_case_id` to `use_case_key` here, since we don't know who's calling this function in what way right now.

Reply:
Are you worried about this breaking keyword arguments calls like
```python
indexer.resolve(use_case_id=..., org_id=..., string=...)
```

Reply:
yeah exactly

Reply:
So maybe let's go back to the original signature for now and change these when we actually alter the indexer interface read path